### PR TITLE
[V3.8.5 pipeline] Add window resize handler.

### PR DESCRIPTION
**Pull request by star-e**
Refactored custom pipeline resize logic.
Previously users need to detect render window resize by themselves. It is tedious and hard to do.
New approach will call user defined resize handler.
New approach will only be enabled, if user defined `gameWindowResize` handler.

### Changelog

* Add resize related members to the `RenderWindow` class.
* Add `colorName`, `depthStencilName` to the `RenderWindow` class. Those are used by custom pipeline.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

File: cocos/render-scene/core/render-window.ts [hunk 1]
```diff
@@ -103,6 +150,14 @@ export class RenderWindow {
     protected _hasOffScreenAttachments = false;
     protected _framebuffer: Framebuffer | null = null;
     protected _device: Device | null = null;
+    // _renderWindowId is used to identify the render window in the render pipeline
+    protected _renderWindowId = _windowCount++;
+    // _isResized is used to indicate whether the render window is resized
+    protected _isResized = true;
```
Comment (minggo) [inline cocos/render-scene/core/render-window.ts:156]: Why default value is true?

Comment (star-e) [reply cocos/render-scene/core/render-window.ts:156]: When render window is initialized, we consider it is resized. So the resize handler can be triggered and user can register resources in the handler.

Comment (minggo) [reply cocos/render-scene/core/render-window.ts:156]: Got it.

**Review by minggo — APPROVED** [review]

**Pull request merged**